ironfish/src/mining/stratum/stratumServer.ts

```diff
@@ -179,19 +184,22 @@ export class StratumServer {
           // removing this undefined check in a future update once we have given enough
           // notice after this deploy.
           if (body.result.version !== undefined && body.result.version < this.versionMin) {
-            this.peers.ban(
-              client,
-              `Client version ${body.result.version} does not meet minimum version ${this.versionMin}`,
-              FIFTEEN_MINUTES_MS,
-            )
+            this.peers.ban(client, {
+              message: `Client version ${body.result.version} does not meet minimum version ${this.versionMin}`,
+              reason: DisconnectReason.BAD_VERSION,
+              until: Date.now() + FIFTEEN_MINUTES_MS,
+              versionExpected: this.version,
+            })
             return
           }
 
           client.publicAddress = body.result.publicAddress
           client.subscribed = true
 
           if (!isValidPublicAddress(client.publicAddress)) {
-            this.peers.ban(client, `Invalid public address: ${client.publicAddress}`)
+            this.peers.ban(client, {
```

Comment:
A ban for this, and some of the other cases feels a bit extreme. It's something easily fixable, and very possible to just typo/copy-paste error. Granted, we have guards on the miner in this case, but if someone is building a different miner and typoes, they now have to wait 15 minutes to try again.

I wonder if a shadow-ban and a disconnect reason would be a better way to deal with a case like this

Reply:
That's a good point, there could be a change here to a less harsh ban but let me bring one more thing up.

The only way this ban can happen is if someone is side stepping the normal client, because this validation happens on the client as well. So if the valid-ness definition diverges, it can also happen. In that case they must be running some custom protocol and are violating protocol.